src/Features/Lsif/Generator/Logging/LsifFormatLogger.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.Writing;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.LanguageServerIndexFormat.Generator.Logging
+{
+    internal sealed class LsifFormatLogger : ILogger
+    {
+        private readonly TextWriter _writer;
+        private readonly object _writerGate = new object();
+
+        public LsifFormatLogger(TextWriter writer)
+        {
+            _writer = writer;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            throw new NotImplementedException();
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return logLevel >= LogLevel.Information;
```

Comment:
Does this mean I need to pass an additional parameter to get telemetry logged?

Reply:
No; ILogger as an interface requires this; this just says this logger doesn't log trace things, since that's not specced as a severity. But we don't have any things doing tracing in the first place.

If you _do_ want some sort of "enable tracing for additional debug logging" we could make this dynamic based on a switch, if you want.

Reply:
Oh, I see. No, that's fine!